File: Python/8py 3/zero_fuel.py
"""
|--------------------------------------------------------------------------
| Will you make it?
|--------------------------------------------------------------------------
|
"""

def zero_fuel(distance_to_pump, mpg, fuel_left):
   return mpg * fuel_left >= distance_to_pump


zero_fuel = lambda _, __, ___: ___ * __ >= _


def zero_fuel(distance_to_pump, mpg, fuel_left):
   return distance_to_pump / mpg <= fuel_left


print(zero_fuel(50, 25, 2), True)
print(zero_fuel(100, 50, 1), False)

File: PHP/8php 3/zeroFuel.php
<?php
/*
|--------------------------------------------------------------------------
| Will you make it?
|--------------------------------------------------------------------------
|
*/

function zeroFuel($distanceToPump, $mpg, $fuelLeft) {
   return $mpg * $fuelLeft >= $distanceToPump;
}


function zeroFuel($distanceToPump, $mpg, $fuelLeft) {
   return $distanceToPump <= $mpg * $fuelLeft;
}


print_r(zeroFuel(50, 25, 2) . ' ' . true . '
');
print_r(zeroFuel(100, 50, 1) . ' ' . false . '
');